tools/EventClients/lib/python/xbmcclient.py

```diff
@@ -249,10 +258,7 @@ def send(self, sock, addr, uid=UNIQUE_IDENTIFICATION):
         """
         self.uid = uid
         for a in range ( 0, self.num_packets() ):
-            try:
```

Comment:
I'd actually like to discuss this change after thinking about it some more.
I do agree with your reasoning but:
 1. that `return True` is now obsolet.
 2. more importantly, there's `kodi-send` which does rely on this and would spew python exception to the user in case of failed connections. Do we want this? Do we care? I'm not sure.

**Edit:**
P.s. No clue where to find the source for the `kodi-send` script.

Reply:
Thanks @hudokkow 
I assume that the comment of mine where I said "Found it" is only visible to me for whatever reason.
I'm so confused by github right now but anyways... For I shall repeat myself :)

[This](https://github.com/xbmc/xbmc/blob/c49d783988a6c1c3adaabfe1425f42e85d3a5ef1/tools/EventClients/Clients/Kodi%20Send/kodi-send.py#L76-L77) is what I was referring to specifically. (Let's not talk about how this file does `from ... import *`)

Reply:
Ah so, when I said approximately zero code is relying on this behaviour, I meant zero code is checking the return value and doing anything more useful with it than raising an exception. Sending a UDP packet can't really fail unless the process has too many file descriptors, or the user doesn't have permissions, or weird things like that. Any exceptions being supressed by this are likely bugs in kodi-send or xbmcclient.py and should be raised as dirty tracebacks just like if any other part of the code failed.

Since kodi-send was not checking the return value, the only change in its behaviour is that if there is a bug in xbmcclient.py or something going wrong with the operating system like ephemeral port exhaustion, it will print a traceback and give a nonzero exit code. This is probably desirable, and is what kodi-send already does in the case of unlikely failures in other parts of the code since exceptions are not suppressed anywhere else.

I actually removed the try: except: in the first place because it was masking bugs in my porting to Python 3.

Reply:
Guess who totally ignored the fact we're talking upd here  _points at self_
Sounds all reasonable and I'm all good :)